hacks/prune-rhcos-snapshots/prune-rhcos-snapshots.py

```diff
@@ -0,0 +1,255 @@
+                    if retry == 4:
+                        raise
+
+            print(f'  Processing installer commit: {installer_commit}')
+            try:
+                gitshow_process = subprocess.run(['git', '-C', str(installer_git_path), 'show', f'{installer_commit}:data/data/coreos/rhcos.json'], capture_output=True)
+                gitshow_process.check_returncode()
+            except:
+                gitshow_process = subprocess.run(['git', '-C', str(installer_git_path), 'show', f'{installer_commit}:data/data/rhcos.json'], capture_output=True)
+                gitshow_process.check_returncode()
+
+            # Otherwise gitshow process contains a file like https://github.com/openshift/installer/blob/01adff5d629bb403a6cf49f9d44070762ef77e93/data/data/coreos/rhcos.json
+            # or older format like https://github.com/openshift/installer/blob/bfdba2d19e3fa1105bab93aa4cce8bb7f44b4b2c/data/data/rhcos.json
+            installer_rhcos_json = json.loads(gitshow_process.stdout)
+
+            installer_ami_ids = dict()
+            if 'amis' in installer_rhcos_json:
+                # Older style without arch
+                for region, region_entry in installer_rhcos_json['amis'].items():
+                    ami_id = region_entry['hvm']
+                    installer_ami_ids[region] = {
+                        'images': [ami_id]
+                    }
+                    must_preserve.add(f'{region}:{ami_id}')
+            else:
+                for installer_arch_name, installer_arch in installer_rhcos_json['architectures'].items():
+                    print(yaml.dump(installer_arch))
+                    if 'images' in installer_arch['artifacts']:
+                        images_list = installer_arch['artifacts']['images']
+                    else:
+                        images_list = installer_arch['images']
+                    if 'aws' not in images_list:
+                        print(f'  No AMIs for arch: {installer_arch_name}')
+                        continue
+                    for region_name, region_entry in images_list['aws']['regions'].items():
+                        if region_name not in installer_ami_ids:
+                            installer_ami_ids[region_name] = {}
+                            installer_ami_ids[region_name]['images'] = []
+                        ami_id = region_entry['image']
+                        installer_ami_ids[region_name]['images'].append(ami_id)
+                        must_preserve.add(f'{region_name}:{ami_id}')
+
+            rhcos_in_use[payload_key] = {
+                'payload_version': machine_os_rhcos_version,
+                'installer_amis': installer_ami_ids
+            }
+
+            with riu_path.open('a+') as f:
+                cache_entry = yaml.dump({
+                    payload_key: rhcos_in_use[payload_key]
+                })
+                # Write, line by line, to the cache file
```

Comment:
minor note that `yaml.dump` includes a line break.

for a moment i was confused thinking each entry would be one line. perhaps better to say:
```suggestion
                # Write to the cache file separated by a blank line
```